cmd/cloudshell_open/docker.go

```diff
@@ -28,11 +33,28 @@ func build(dir, image string) error {
 	return nil
 }
 
-func push(image string) error {
+func dockerPush(image string) (err error) {
+	fmt.Println(infoPrefix + " FYI, running the following command:")
+	cmdColor.Printf("\tdocker push %s\n", parameter(image))
+	end := logProgress("Pushing container image...",
+		"Pushed container image to Google Container Registry.",
+		"Failed to push container image to Google Container Registry.")
+	defer func() {
+		end(err == nil)
+	}()
+
 	cmd := exec.Command("docker", "push", image)
 	b, err := cmd.CombinedOutput()
 	if err != nil {
 		return fmt.Errorf("docker push failed: %v, output:\n%s", err, string(b))
 	}
 	return nil
 }
+
+func dockerFileExists(dir string) bool {
+	if _, err := os.Stat(filepath.Join(dir, "Dockerfile")); err != nil {
```

Comment:
Err != nil is a bad check here
This  can fail for other reasons. Need to test:
1. Is file
2. Err is ENOENT.

Reply:
Done